ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -251,6 +258,26 @@ export default class ConfirmTransactionBase extends Component {
               primaryValueTextColor="#2f9ae0"
             />
           </div>
+          {useNonceField ? <div>
+            <div className="confirm-detail-row">
+              <div className="confirm-detail-row__label">
+                { this.context.t('nonceFieldHeading') }
+              </div>
+              <div className="custom-nonce-input">
+                <TextField
+                  type="number"
+                  min="0"
+                  placeholder={ this.context.t('nonceFieldPlaceholder') }
+                  onChange={({ target: { value }}) => {
+                    updateCustomNonce(value < 0 ? 0 : Math.floor(value))
```

Comment:
The custom nonce is a string everywhere else; it might be better to cast it to a string here as well. Better to avoid it having multiple types if possible, just to keep things simpler.